telegram/tests/test_telegram.py

```diff
@@ -0,0 +1,201 @@
+        returnValue(transport)
+
+    def handle_inbound_request(self, req):
+        self.request_queue.put(req)
+        return NOT_DONE_YET
+
+    @inlineCallbacks
+    def get_next_request(self):
+        req = yield self.request_queue.get()
+        self.pending_requests.append(req)
+        returnValue(req)
+
+    @inlineCallbacks
+    def finish_requests(self):
+        for req in self.pending_requests:
+            if not req.finished:
+                yield req.finish()
+
+    def test_translate_inbound_message_from_channel(self):
+        default_channel = {
+            'id': 'Default channel',
+            'type': 'channel',
+        }
+        inbound_msg = {
+            'message_id': 'Message from Telegram channel',
+            'chat': default_channel,
+            'text': 'Hi from Telegram channel!',
+        }
+
+        message = self.transport.translate_inbound_message(inbound_msg)
+        self.assertEqual(inbound_msg['text'], message['content'])
+        self.assertEqual(self.bot_username, message['to_addr'])
+        self.assertEqual(default_channel['id'], message['from_addr'])
+
+    def test_translate_inbound_message_from_user(self):
+        inbound_msg = {
+            'message_id': 'Message from Telegram user',
+            'chat': 'Random chat',
+            'text': 'Hi from Telegram user!',
+            'from': self.default_user,
+        }
+
+        message = self.transport.translate_inbound_message(inbound_msg)
+        self.assertEqual(inbound_msg['text'], message['content'])
+        self.assertEqual(self.bot_username, message['to_addr'])
+        self.assertEqual(self.default_user['id'], message['from_addr'])
+
+    def test_translate_inbound_message_no_text(self):
+        inbound_msg = {
+            'message_id': 'Message without text',
+            'chat': 'Random chat',
+            'from': self.default_user,
+        }
+
+        message = self.transport.translate_inbound_message(inbound_msg)
+        self.assertEqual('', message['content'])
+        self.assertEqual(self.bot_username, message['to_addr'])
+        self.assertEqual(self.default_user['id'], message['from_addr'])
+
+    @inlineCallbacks
+    def test_inbound_update(self):
+        default_update = json.dumps({
+            'update_id': 'update_id',
+            'message': {
+                'message_id': 'msg_id',
+                'from': self.default_user,
+                'chat': {
+                    'id': 'chat_id',
+                    'type': self.PRIVATE
+                },
+                'date': 1234,
+                'text': 'Incoming message from Telegram!',
+            }
+        })
+        res = yield self.helper.mk_request(_method='POST',
+                                           _data=default_update)
+        self.assertEqual(res.code, 200)
+
+        [msg] = yield self.helper.wait_for_dispatched_inbound(1)
+
+        expected_update = json.loads(default_update)
```

Comment:
We seem to be doing a `json.loads()` of something we `json.dumps()` a few lines up. Maybe we should instead declare `default_update` as a dict, use `json.dumps()` it where we need to, and use `default_update` where we are using `expected_update`.